libs/core/src/components/spinner/gds-spinner.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * @fileoverview GDS Spinner Component - A loading indicator with accessibility support
+ * @module gds-spinner
+ */
+
+import { css } from 'lit'
+import { property } from 'lit/decorators.js'
+
+import { GdsElement } from '../../gds-element'
+import { tokens } from '../../tokens.style'
+import { TransitionalStyles } from '../../transitional-styles'
+import {
+  gdsCustomElement,
+  html,
+} from '../../utils/helpers/custom-element-scoping'
+import { styles } from './gds-spinner.styles'
+
+/**
+ * GdsSpinner Component
+ * A loading indicator web component that supports various sizes and display modes.
+ *
+ * @extends {GdsElement}
+ *
+ * @example
+ * ```html
+ * <!-- Basic usage -->
+ * <gds-spinner size="md" text="Loading..."></gds-spinner>
+ *
+ * <!-- Screen reader only text -->
+ * <gds-spinner size="md" text="Loading..." .visualText=${false}></gds-spinner>
+ *
+ * <!-- Cover container -->
+ * <gds-spinner size="md" text="Loading..." cover></gds-spinner>
+ * ```
+ *
+ * @csspart spinner - The spinning circle element
+ * @csspart text - The loading text element
+ *
+ * @fires gds-element-disconnected - Fired when the element is removed from the DOM
+ */
+@gdsCustomElement('gds-spinner')
+export class GdsSpinner extends GdsElement {
+  /**
+   * Component styles combining design tokens, host styles, and component-specific styles
+   * @type {Array<CSSStyleSheet>}
+   */
+  static styles = [
+    tokens,
+    css`
+      :host {
+        display: block;
+      }
+    `,
+    styles,
+  ]
+
+  /**
+   * Loading text to be displayed. Can be shown visually or for screen readers only.
+   * @type {string|undefined}
+   */
+  @property({ type: String })
+  text?: string
+
+  /**
+   * Controls text visibility mode.
+   * When true, text is visible and announced to screen readers.
+   * When false, text is only announced to screen readers.
+   * @type {boolean}
```

Comment:
Type information is not needed in doc comments, since typescript can already infer this from the code.